scipy/signal/tests/test_windows.py

```diff
@@ -538,4 +539,9 @@ def test_needs_params():
                    'slepian', 'optimal', 'slep', 'dss', 'dpss',
                    'chebwin', 'cheb', 'exponential', 'poisson', 'tukey',
                    'tuk']:
-        assert_raises(ValueError, signal.get_window, winstr, 7)
+        assert_raises(ValueError, get_window, winstr, 7)
+
+
+def test_deprecation():
+    assert 'signal.hann is deprecated' in dep_hann.__doc__
```

Comment:
Odd, this test should not be passing on TravisCI with ``OPTIMIZE=-OO``. That is run under Python 3.4, and I've checked that it picks up ``-OO``, because it says at the top of the test output

    WARNING: assertions not in test modules or plugins will be ignored because assert 
    statements are not executed by the underlying Python interpreter (are you using python -O?)

It does fail as expected for me on Python 2.7. I'll push a commit moving the ``-OO`` run to 2.7

Reply:
Note that there are 2 issues here, use of ``assert`` and use of ``__doc__``

Reply:
Good points, should be fixed now (your commit broke Travis as intended)

Reply:
No it didn't break, that's the ``special`` failures with numpy master. I added ``-OO`` to the wrong build matrix entry.

Reply:
I'll remove my commit and then merge. Will figure out ``-OO`` some other time, opening a new issue for it.

Reply:
Here's the link to the TravisCI run (the link to it will disappear from this PR once I force-push): https://travis-ci.org/scipy/scipy/builds/278796885?utm_source=github_status&utm_medium=notification

Reply:
Issue opened: gh-7914